overlord/ifacestate/ifacemgr.go

```diff
@@ -22,36 +22,91 @@
 package ifacestate
 
 import (
+	"github.com/ubuntu-core/snappy/interfaces"
+	"github.com/ubuntu-core/snappy/interfaces/builtin"
 	"github.com/ubuntu-core/snappy/overlord/state"
 )
 
 // InterfaceManager is responsible for the maintenance of interfaces in
 // the system state.  It maintains interface connections, and also observes
 // installed snaps to track the current set of available plugs and slots.
-type InterfaceManager struct{}
+type InterfaceManager struct {
+	// state holds a reference to persistent system state. Interface engine
+	// uses the state to store list of "intents-to-connect" that were made by
+	// the operator. Those intents are made effective during Ensure().
+	state *state.State
+	// repo contains all the interfaces, plugs and slots declared by snaps (or
+	// added at runtime via the snap experimental command) and all the
+	// connection made between plugs and slots.
+	//
+	// Repository is where the volatile runtime state is held. When a snaps are
+	// installed, removed or changed in any way (updated) the repository must
+	// be kept up-to-date as to what slots and plugs exist in the system.
+	repo *interfaces.Repository
+	// uDevRulesChanged is a flag monitoring if udev rules have been modified
+	// on disk and need to be re-loaded by udev (with an explicit request).
+	// Udev rules are re-loaded once, after making all the changes to udev
+	// rules.
+	uDevRulesChanged bool
+	// aaProfilesLoaded is a set of all the loaded apparmor profiles. Profiles
+	// are kept in kernel memory and are not persistent across reboots. This
+	// set is here to help us purge unused profiles at runtime.
+	//
+	// NOTE: there's a systemd job that loads all existing profiles from
+	// snappy-specific var directory early on boot. Those are not reflected
```

Comment:
We could discard that and load everything ourselves or we could assume they got loaded and simply ignore them. They will be removed on the first change of the state, as explained in the comment above.